sip-viewer/src/app/session-table/session-table.component.ts

```diff
@@ -97,34 +101,33 @@ export class SessionTableComponent implements AfterViewInit {
         this.dataSource = new MatTableDataSource(this.tableData);
 
         // Select all sessions when the application is started
-        this.dataService.updateSelectedSessionsByList(this.sessionIDList);
-        this.sessionIDsToSendToDataService = this.sessionIDList;
-        this.dataSource.data.forEach((row) => {
-          this.selection.select(row);
-        });
+        this.selection.select(...this.dataSource.data);
+        this.dataService.updateSelectedSessionsByList(
+          this.getSelectedSessions()
+        );
       },
       (error) => {
         console.error('Error fetching messages', error);
       }
     );
   }
 
+  getSelectedSessions(): any {
+    const selectedSessions: any[] = [];
+    this.selection.selected.forEach((session) => {
+      selectedSessions.push(session);
+    });
+    return selectedSessions;
+  }
+
   onCheckboxClicked(row: any): void {
-    this.selection.select(row);
-    const sessionID = row.SessionID;
-    if (this.sessionIDsToSendToDataService.indexOf(sessionID) !== -1) {
-      let index = this.sessionIDsToSendToDataService.indexOf(sessionID);
-      this.sessionIDsToSendToDataService.splice(index, 1);
-    } else {
-      this.sessionIDsToSendToDataService.push(sessionID);
-    }
-    this.dataService.updateSelectedSessionsByList(
-      this.sessionIDsToSendToDataService
-    );
-    if (this.sessionIDsToSendToDataService.length === 0) {
-      this.headerCheckbox.toggle();
-      this.selection.clear();
+    const selectedSessions = this.getSelectedSessions();
+    if (selectedSessions.includes(row)) {
+      this.selection.deselect(row);
+    } else if (!selectedSessions.includes(row)) {
```

Comment:
Check if else if is necessary. If not, then you can just change it to else. This makes the code much more readable